ce/api/streamflow/watershed.py

```diff
@@ -123,6 +123,8 @@ def worker(station_lonlat, flow_direction, elevation, area, hypso_params=None):
         raise ValueError("Flow direction and elevation do not have compatible grids")
     if not flow_direction.is_compatible(area):
         raise ValueError("Flow direction and area do not have compatible grids")
+    if elevation.units != "m" or area.units != "m2":
+        raise ValueError("Elevation and area do not have compatible units")
```

Comment:
Quibble: Either the condition is too strict (only `m` as the length unit) or the error message is too general. For example, `yd` and `yd^2` are compatible but would raise this error.

Reply:
My personal inclination would be to use a units-parsing utility and do a more general comparison. Unless `m` really is  the only acceptable unit. However, that's more effort.

Reply:
That was my first line of attack, but `pint` doesn't understand `m2`!

Reply:
Or pass run the string through a search/replace that inserts a `^` before digit strings.